### DISCUSSION TITLE
Is it possible to install lefthook into the global git template directory?

### DISCUSSION THREAD

**Discussion by nogweii** [evilmartians/lefthook#746] [Q&A]
Assuming that I have a number of repositories each with a lefthook configuration file, at the moment I have to remember to run `lefthook install` in each clone. Git has [a template directory configured with `init.templateDir`](https://git-scm.com/docs/git-init#_template_directory) that it will copy the contents of into every freshly cloned repository.

Is there a set of generic hook scripts that can be made to automatically run `lefthook install` in repos that have it configured while safely ignoring any repos that don't have it configured?

**Comment by mrexox** [comment]
If I understand it correctly, the templteDir provides a template for the `.git` directory, not for the root dir. In this case the `lefthook.yml` is not being added. Without the `lefthook.yml` hooks don't make sense even if they are already installed into `.git/hooks`

If you want `lefthook install` to run automatically consider using `npm` package [`lefthook`](https://www.npmjs.com/package/lefthook) which auto-installs when you install package dependencies.

Unfortunately, there is no other way right now to do auto-install on git-clone 😕

**Reply by nogweii** [reply]
Yeah, the contents of the directory specified will be copied into `.git`. Including the `hooks` subfolder.

So I was hoping that `lefthook` could produce a series of hook scripts that would check for the presence of that `lefthook.yml` file and execute lefthook as needed. Basically, every repo would have a bunch of scripts that would forward all of the parameters to lefthook only if the binary was found in `$PATH`.

I guess running `lefthook add` for every hook git has and copying the resulting scripts into the template directory is pretty close to what I'm looking for? I think that would be pretty safe, after editing the generated scripts to execute lefthook only if the configuration file was found.

**Reply by mrexox** [reply]
Probably using scripts and storing them in `.git/` folder is a solution. After that you'll just need a configuration file `lefthook.yml` with the scripts specified. This could work :+1:

However this is not the idea of lefthook to keep scripts in a hidden `.git/` folder. Lefthook scripts are supposed to be committed with a repo code.
